src/core/search/search.h

```diff
@@ -84,8 +94,8 @@ class FieldIndices {
   void CreateIndices(PMR_NS::memory_resource* mr);
   void CreateSortIndices(PMR_NS::memory_resource* mr);
 
- private:
-  Schema schema_;
+  const Schema* schema_;
+  const IndicesOptions* options_;
```

Comment:
Same shared_ptr vs ptr question here. Everything points esentially to shared_ptr<DocIndex> that is created in the coordinator when parsing the index definition. Lifetimes are ensured by ShardDocIndex that create the FieldIndices

Reply:
if you feel that `shared_ptr` is more correct because of lack of single owner, then lets switch to shared_ptr